src/ape_ethereum/ecosystem.py

```diff
@@ -320,11 +325,69 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
 
         return txn_class(**kwargs)  # type: ignore
 
-    def decode_event(self, abi: EventABI, receipt: "ReceiptAPI") -> "ContractLog":
-        filter_id = keccak(to_bytes(text=abi.selector))
-        event_data = next(log for log in receipt.logs if log["filter_id"] == filter_id)
+    def decode_logs(self, abi: EventABI, data: List[Dict]) -> Iterator["ContractLog"]:
```

Comment:
Part of me wonders if `ContractLog` would be better off in `ape.types`

Reply:
I am inclined to agree, as it will import much nicer everywhere it is used this way.
Also, since Evan thumbs up, I assume he approves of that change as well.